### PR TITLE
[Collapsible] Add inline variant 

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris-backlog/issues/1606

While prototyping multiple options for x-axis animations I realized I was getting into a lot of what collapsible already does.

TIL: `transform` animations don't affect layouts, which is why they're less expensive but won't work in our use case.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by laurkim — APPROVED** [review]
Nice!! 💯 🙌

**Review by sophschneider — APPROVED** [review]
cool!

**Review by sophschneider — APPROVED** [review]
new commit LGTM

File: .changeset/strong-experts-remember.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Added horizontal variant to Collapsible
```
Comment (kyledurand) [inline .changeset/strong-experts-remember.md]:
```suggestion
Added `variant` prop to Collapsible
```

**Pull request merged**